### ISSUE TITLE
selecting data endpoint UoL

### ISSUE THREAD

**Issue by opeeters** [52North/sensorweby#21]
Selecting data from this endpoint: www57.lamp.le.ac.uk/52n-sos-webapp/api/v1
results in the error below
a bit strange, since this call:
http://www57.lamp.le.ac.uk/52n-sos-webapp/api/v1/categories
does return categories correctly
Is this also a cross-domain issue?

_______
```
"Publish timeseries:data:load" jsc-1.0.0-SNAPSHOT.js:1847:0
"
Listening on http://127.0.0.1:46012

Attaching package: ‘sensorweby’

The following object is masked from ‘package:shiny’:

    runExample

Welcome to sensorweb4R! The logging level is INFO - set it with 'flog.threshold(<level>, name = "sensorweb4R")'
Loading required package: lazyeval
Loading required package: dplyr

Attaching package: ‘dplyr’

The following object is masked from ‘package:sensorweb4R’:

    id

The following object is masked from ‘package:stats’:

    filter

The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union

Loading required package: maps
Error in value[[3L]](cond) : 
  Error requesting 'http://www57.lamp.le.ac.uk/52n-sos-webapp/api/v1/categories/NULL?': Error in doTryCatch(return(expr), name, parentenv, handler): server error: (500) Internal Server Error

{"statusCode":500,"reason":"Internal Server Error","userMessage":"Could not get category data."}
" shiny-server.js:210:46
"Publish timeseries:data:loadfinished" jsc-1.0.0-SNAPSHOT.js:1847:0
"Publish timeseries:synced"
```

**Comment by autermann** [comment]
No timeseries has a category (see [`http://www57.lamp.le.ac.uk/52n-sos-webapp/api/v1/timeseries?expanded=true`](http://www57.lamp.le.ac.uk/52n-sos-webapp/api/v1/timeseries?expanded=true):
```json
{
    "id": "69",
    "label": "air-temp campbell-ws-cr200, http://uk-air.defra.gov.uk/networks/site-info?site_id=LECU",
    "uom": "°C",
    "station": {
      "properties": {
        "id": 7,
        "label": "http://uk-air.defra.gov.uk/networks/site-info?site_id=LECU"
      },
      "geometry": {
        "coordinates": [
          -1.1273109912872314,
          52.61982345581055
        ],
        "type": "Point"
      },
      "type": "Feature"
    },
    "referenceValues": [],
    "firstValue": {
      "timestamp": 1416652860000,
      "value": 13.36
    },
    "lastValue": {
      "timestamp": 1426809720000,
      "value": 5.56
    },
    "parameters": {
      "service": {
        "id": "1",
        "label": "University of Leicester: Sensor Observation Service."
      },
      "offering": {
        "id": "28",
        "label": "campbell-ws-cr200"
      },
      "feature": {
        "id": "7",
        "label": "http://uk-air.defra.gov.uk/networks/site-info?site_id=LECU"
      },
      "procedure": {
        "id": "28",
        "label": "campbell-ws-cr200"
      },
      "phenomenon": {
        "id": "84",
        "label": "air-temp"
      }
    }
  }
```
Is that intended? In any case we should add a simple check to compensate this.

**Comment by nuest** [comment]
+1 for the check that this does not cause any errors.

**Comment by opeeters** [comment]
could this be due to the fact that in server.R the variable "endpoint" is defined as:
endpoint <- example.endpoints()[2]
Hence only the details of endpoint 2 of example.endpoints() is available?

**Comment by autermann** [comment]
> endpoint <- example.endpoints()[2]

Which file is that?

**Comment by opeeters** [comment]
all example server.R, e.g. https://github.com/52North/sensorweby/blob/master/inst/examples/pollutionPlot/server.R

**Comment by opeeters** [comment]
additionally, I find this in the console output of R-studio when I select e.g. 2 time series from http://sosrest.irceline.be/api/v1 and one from http://sensors.geonovum.nl/sos/api/v1:


[INFO] [2015-03-27 12:56:35] [sensorweby.::] Parsing timeseries:
	http://sosrest.irceline.be/api/v1/timeseries/ts_cde47235e7e66c346fac4d47d84a5986
	http://sosrest.irceline.be/api/v1/timeseries/ts_58b06b4fd39878dcc38b76a8c2cbd0ad
	http://sensors.geonovum.nl/sos/api/v1/timeseries/141
Warning in (function (FUN, ..., MoreArgs = NULL, SIMPLIFY = TRUE, USE.NAMES = TRUE)  :
  longer argument not a multiple of length of shorter
Error in (function (classes, fdef, mtable)  : 
  unable to find an inherited method for function ‘supportsFirstLatest<-’ for signature ‘"Service", "list"’
In addition: Warning messages:
1: In gsub("~f", the.function, message, fixed = TRUE) :
  argument 'replacement' has length > 1 and only the first element will be used
2: In gsub("~f", the.function, message, fixed = TRUE) :
  argument 'replacement' has length > 1 and only the first element will be used

**Comment by opeeters** [comment]
I should have posted this here maybe: https://github.com/52North/sensorweby/issues/20
There might be a connection. Maybe function "supportsFirstLatest" is expecting a reponse differently than what the geonovum endpoint is returning.

**Comment by autermann** [comment]
> all example server.R, e.g. https://github.com/52North/sensorweby/blob/master/inst/examples/pollutionPlot/server.R

well... the pollution plot won't work with other APIs/services as it requires wind data that only can be found in the IRCELINE endpoint (also the distance matrix to get the nearest station is only build for this single service).

And I can't find it in e.g. `timePlot`.


> additionally, I find this in the console output of R-studio when I select e.g. 2 time series from http://sosrest.irceline.be/api/v1 and one from http://sensors.geonovum.nl/sos/api/v1:
> 
> [INFO] [2015-03-27 12:56:35] [sensorweby.::] Parsing timeseries:
> http://sosrest.irceline.be/api/v1/timeseries/ts_cde47235e7e66c346fac4d47d84a5986
> http://sosrest.irceline.be/api/v1/timeseries/ts_58b06b4fd39878dcc38b76a8c2cbd0ad
> http://sensors.geonovum.nl/sos/api/v1/timeseries/141
> Warning in (function (FUN, ..., MoreArgs = NULL, SIMPLIFY = TRUE, USE.NAMES = TRUE) :
> longer argument not a multiple of length of shorter
> Error in (function (classes, fdef, mtable) : 
> unable to find an inherited method for function ‘supportsFirstLatest<-’ for signature ‘"Service", "list"’

I thought I've fixed that in https://github.com/52North/sensorweb4R/commit/4aef2965d3ad4c9942519fb6144d911ea840f088. Could you update the `sensorweb4R` package?

> In addition: Warning messages:
> 1: In gsub("~f", the.function, message, fixed = TRUE) :
> argument 'replacement' has length > 1 and only the first element will be used
> 2: In gsub("~f", the.function, message, fixed = TRUE) :
> argument 'replacement' has length > 1 and only the first element will be used

Warnings related to logging…

**Comment by opeeters** [comment]
OK, excellent. The error above is indeed taken care of with https://github.com/52North/sensorweb4R/commit/4aef2965d3ad4c9942519fb6144d911ea840f088. I get a diffente error now, but I will take te discussion to https://github.com/52North/sensorweby/issues/20 now

**Comment by nuest** [comment]
@opeeters so we can close this issue, correct?

**Comment by autermann** [comment]
Fixed in 52North/sensorweb4R#24
